MaintenanceModePlugin.inc.php

```diff
@@ -52,13 +51,87 @@ public function getDescription()
     }
 
+    /**
+     * @copydoc Plugin::getActions()
+     */
+    public function getActions($request, $verb)
+    {
+        $router = $request->getRouter();
+        import('lib.pkp.classes.linkAction.request.AjaxModal');
+        return array_merge(
+            $this->getEnabled()?array(
+                new LinkAction(
+                    'settings',
+                    new AjaxModal(
+                        $router->url($request, null, null, 'manage', null, array('verb' => 'settings', 'plugin' => $this->getName(), 'category' => 'generic')),
+                        $this->getDisplayName()
+                    ),
+                    __('manager.plugins.settings'),
+                    null
+                ),
+            ):array(),
+            parent::getActions($request, $verb)
+        );
+    }
+
+    /**
+     * @copydoc Plugin::manage()
+     */
+    public function manage($args, $request) {
+        switch ($request->getUserVar('verb')) {
+            case 'settings':
+                AppLocale::requireComponents(LOCALE_COMPONENT_APP_COMMON,  LOCALE_COMPONENT_PKP_MANAGER);
+                $templateMgr = TemplateManager::getManager($request);
+                $templateMgr->registerPlugin('function', 'plugin_url', array($this, 'smartyPluginUrl'));
+
+                $this->import('classes.MaintenanceModeSettingsForm');
+                $form = new MaintenanceModeSettingsForm($this);
+
+                if ($request->getUserVar('save')) {
+                    $form->readInputData();
+                    if ($form->validate()) {
+                        $form->execute();
+                        return new JSONMessage(true);
+                    }
+                } else {
+                    $form->initData();
+                }
+                return new JSONMessage(true, $form->fetch($request));
+        }
+        return parent::manage($args, $request);
+    }
+
     protected function maintenanceModeChecker(): void
     {
         $file = Core::getBaseDir() . DIRECTORY_SEPARATOR . '.maintenance';
 
         if (file_exists($file)) {
-            header('HTTP/1.1 503 Service Temporarily Unavailable');
+            $useCustomUrl = $this->getSetting(CONTEXT_SITE, 'useCustomUrl');
+            $customUrl = $this->getSetting(CONTEXT_SITE, 'customUrl');
+
+            if ($useCustomUrl && $this->customUrlIsValid($customUrl)) {
+                header('Location: ' . $customUrl, true, 307);
+            } else {
+                header('HTTP/1.1 503 Service Temporarily Unavailable');
+            }
             die(__('plugins.generic.maintenancemode.underMaintenance'));
         }
     }
+
+    /**
+     * Check if desired redirect is uses base path that's being redirected
+     *
+     * @param string|null $customUrl
+     * @return bool
+     */
+    public function customUrlIsValid(?string $customUrl): bool
+    {
+        $baseUrl = \Config::getVar('general', 'base_url');
```

Comment:
What do you think in inject the dependency with `use PKP\config\Config`?

Reply:
I don't think that will work with OJS 3.3 as I think the namespacing work came after.

Reply:
You're correct!